pennylane/utils.py

```diff
@@ -306,6 +306,9 @@ def circuit2():
             "This could happen if inversion of a template function is attempted. "
             "Please use inv on the function including its arguments, as in inv(template(args))."
         )
+    elif isinstance(operation_list, qml.tape.QuantumTape):
+        operation_list.inv()
+        return operation_list
```

Comment:
Elegant! I am so confused here though. Above don't you make sure the templates decorator still uses the `OperationRecorder`? But here it seems a template is a tape? Wait, let me try to figure it out by reading again above...

Reply:
Oh wait, you do change the logic for templates recorded with `OperationRecorders` below. When could the case above happen then? Is it just to make the function more powerful?